drivers/counter/counter_mspm0.c

```diff
@@ -107,8 +107,8 @@ static uint32_t counter_mspm0_get_freq(const struct device *dev)
 	struct counter_mspm0_data *data = dev->data;
 	const struct counter_mspm0_config *cfg = dev->config;
 	const DL_Timer_ClockConfig clock_cfg = cfg->clock_cfg;
-	// freq = (timer_clk_source / (div_ratio * (prescale + 1))
-	return data->freq / (clock_cfg.divideRatio * (clock_cfg.prescale + 1));
+	// freq = (timer_clk_source / ((div_ratio + 1) * (prescale + 1))
```

Comment:
I guess this comment is still needed for testing purposes?

Reply:
still the timers are behaving a bit funny. so trying to keep everything detailed as much as possible until i iron everything out